### DISCUSSION TITLE
.m3u8 url for YouTube and Kick

### DISCUSSION THREAD

**Discussion by Shivansh-yadav13** [streamlink/streamlink#5580] [Q&A] [answered]
### Description

I'm using streamlink to get the .m3u8 url for twitch VODs,
is there a way to get the same for YouTube and Kick

```py
streams = streamlink.streams(url)
```

I'm not able to find anything about the Kick platform, I found few plugins made for kick but they don't have much docs and I think they are for command line (not sure).
I just want to get the .m3u8 url

**Comment by bastimeyer** [comment, marked as answer]
> I'm not able to find anything about the Kick platform

Streamlink does not have a plugin for that site, for various reasons. You won't find support here for third party plugins.

> I think they are for command line

The job of a plugin is to implement Streamlink's plugin API, so that the Streamlink session can retrieve the streams. The retrieved streams are then presented to the user via Streamlink's CLI. However, if you're using Streamlink's Python API in your Python application, then you should take a look at the documentation, because Streamlink's CLI won't be of use for you.

- https://streamlink.github.io/api.html

> I just want to get the .m3u8 url

Streams must be `HLSStream` instances. Muxed streams or other stream types can't or don't necessarily have specific stream URLs.

- https://streamlink.github.io/api/stream.html#streamlink.stream.HLSStream
- https://streamlink.github.io/api/stream.html#streamlink.stream.HTTPStream.url

> YouTube

Streamlink's YouTube plugin does only return `HLSStream`s for live streams. Everything else is a muxed stream.

----

```sh
$ streamlink --json twitch.tv/eslcs | jq '.streams | map_values(.url)'
```
```json
{
  "audio_only": "https://...",
  "160p": "https://...",
  "360p": "https://...",
  "480p": "https://...",
  "720p": "https://...",
  "720p60": "https://...",
  "1080p60": "https://...",
  "worst": "https://...",
  "best": "https://..."
}
```

```python
>>> from streamlink import Streamlink
>>> from streamlink.stream import HLSStream
>>> 
>>> session = Streamlink()
>>> streams = session.streams("twitch.tv/eslcs")
>>> {name: stream.url for name, stream in streams.items() if isinstance(stream, HLSStream)}
{'audio_only': 'https://...', '160p': 'https://...', '360p': 'https://...', '480p': 'https://...', '720p': 'https://...', '720p60': 'https://...', '1080p60': 'https://...', 'worst': 'https://...', 'best': 'https://...'}
```

**Comment by BSClient** [comment]
@Shivansh-yadav13 do you know how to generate a link on the kick.com platform during a broadcast

**Comment by BSClient** [comment]
please program downloader

**Comment by BSClient** [comment]
@bastimeyer how do I run what you wrote
